test/utils/cluster.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const listFormat = "value(NAME)"
+
+func EnsureCluster(clusterName, zone string, numOfNodes int) error {
+	if !isClusterExisting(clusterName, zone) {
+		klog.Infof("Cluster %s in zone %s does not exist, creating.", clusterName, zone)
+		return createCluster(clusterName, zone, numOfNodes)
+	}
+	params := []string{
+		"container",
+		"clusters",
+		"describe",
+		clusterName,
+		"--zone", zone,
+	}
+	out, err := exec.Command("gcloud", params...).CombinedOutput()
```

Comment:
David -- this flow doesn't make sense.

* If the cluster doesn't exist, why do we want to describe it.
* If describe returns an error, why do we want to try deleting a cluster that probably doesn't exist?

Generally if you get an error, you should bail and exit with an error code (klog.Fatalxxx will do this)

Reply:
> If the cluster doesn't exist, why do we want to describe it.
The previous message should be change to `gcloud describe` failed.
> If describe returns an error, why do we want to try deleting a cluster that probably doesn't exist?
Updated. Now if describe fails, we exit.